src/harness/virtualFileSystemWithWatch.ts

```diff
@@ -279,151 +261,68 @@ interface Array<T> { length: number; [n: number]: T; }`
         }
     }
 
-    class Callbacks {
-        private map: TimeOutCallback[] = [];
-        private nextId = 1;
-
-        getNextId() {
-            return this.nextId;
-        }
-
-        register(cb: (...args: any[]) => void, args: any[]) {
-            const timeoutId = this.nextId;
-            this.nextId++;
-            this.map[timeoutId] = cb.bind(/*this*/ undefined, ...args);
-            return timeoutId;
-        }
-
-        unregister(id: any) {
-            if (typeof id === "number") {
-                delete this.map[id];
-            }
-        }
-
-        count() {
-            let n = 0;
-            for (const _ in this.map) {
-                n++;
-            }
-            return n;
-        }
-
-        invoke(invokeKey?: number) {
-            if (invokeKey) {
-                this.map[invokeKey]();
-                delete this.map[invokeKey];
-                return;
-            }
-
-            // Note: invoking a callback may result in new callbacks been queued,
-            // so do not clear the entire callback list regardless. Only remove the
-            // ones we have invoked.
-            for (const key in this.map) {
-                this.map[key]();
-                delete this.map[key];
-            }
-        }
-    }
-
-    type TimeOutCallback = () => any;
-
-    export interface TestFileWatcher {
-        cb: FileWatcherCallback;
-        fileName: string;
-        pollingInterval: PollingInterval;
-    }
-
-    export interface TestFsWatcher {
-        cb: FsWatchCallback;
-        directoryName: string;
-        fallbackPollingInterval: PollingInterval;
-        fallbackOptions: WatchOptions | undefined;
-    }
-
-    export interface ReloadWatchInvokeOptions {
-        /** Invokes the directory watcher for the parent instead of the file changed */
-        invokeDirectoryWatcherInsteadOfFileChanged: boolean;
-        /** When new file is created, do not invoke watches for it */
-        ignoreWatchInvokedWithTriggerAsFileCreate: boolean;
-        /** Invoke the file delete, followed by create instead of file changed */
-        invokeFileDeleteCreateAsPartInsteadOfChange: boolean;
-    }
+    export type TestServerHostTrackingWrittenFiles = TestServerHost & { writtenFiles: ESMap<Path, number>; };
 
-    export enum Tsc_WatchFile {
-        DynamicPolling = "DynamicPriorityPolling",
-        SingleFileWatcherPerName = "SingleFileWatcherPerName"
+    export function changeToHostTrackingWrittenFiles(inputHost: TestServerHost) {

```

Comment:
Can you please move some of these functions and methods to their original location so that the order isnt changed so its easier to review the change. Thanks